### DISCUSSION TITLE
Filament shield

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#3032] [Help] [answered]
Hi all

I installed the filament shield plugin after following the instructions on https://github.com/bezhanSalleh/filament-shield
The role management is well integrated but i don't know how i can attribute a role on users. 
There is no explanation on how associate a role on user except 

Add the Spatie\Permission\Traits\HasRoles trait to your User model(s):
use Illuminate\Foundation\Auth\User as Authenticatable;
use Spatie\Permission\Traits\HasRoles;

class User extends Authenticatable
{
    use HasRoles;
  //.....
}

![roles](https://user-images.githubusercontent.com/58070424/177332573-fb09399d-186a-4ec1-b7e9-1a72b7021ff9.jpg)
![users](https://user-images.githubusercontent.com/58070424/177333179-48ebc38b-55af-4e21-8165-371094cf15f6.jpg)

**Comment by danharrin** [comment]
Open a discussion on the plugin repo, not here

**Reply by eaziweb95** [reply]
ok sorry

**Comment by eaziweb95** [comment]
No answer on the plugin repo ...

**Comment by invaders-xx** [comment, marked as answer]
I don't know if you found your solution, just in case, add this to your user's form:
```php
Forms\Components\Select::make('roles')
                            ->multiple()
                            ->required()
                            ->relationship('roles', 'name'),
```

**Reply by eaziweb95** [reply]
This code works 
Forms\Components\BelongsToManyMultiSelect::make('roles')->relationship('roles', 'name')->label('Rôle')->required(),

**Reply by danharrin** [reply]
It's the same thing
